src/models/comment.rs

```diff
@@ -0,0 +1,63 @@
+use std::error::Error;
+use std::fmt;
+
+#[derive(PartialEq, Debug, Clone, Hash, Eq)]
+/// Represents a comment on a [Ticket](Ticket)
+/// Wraps a string and checks that it is not empty when set
+pub struct Comment {
+    pub comment: String,
```

Comment:
With this structure, it's still possible for a user to ignore your `new` constructor and create a `Comment` using an empty string:
```rust
Comment { comment: "" }
```
It would be better to either make `comment` private or use a wrapper type around `String`, as we did for `Title` in `Ticket`.

Reply:
Cool. I've removed the pub modifier. Not sure why I put that in there as initial revision didn't have it.